apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -1117,6 +1120,26 @@ impl Telemetry {
         TextMapCompositePropagator::new(propagators)
     }
 
+    pub(crate) fn validate_propagation_compatibility(config: &config::Conf) -> Result<(), BoxError> {
+        let propagation = &config.exporters.tracing.propagation;
+        let tracing = &config.exporters.tracing;
+
+        // Check if both Datadog and trace context propagation are active
+        let datadog_active = propagation.datadog || tracing.datadog.enabled();
+        let trace_context_active = propagation.trace_context || tracing.otlp.enabled;
```

Comment:
2: Is also in any similar way true that `tracing.otlp` by default will implicitly enable the `trace_context` propagator?